sentry-android-replay/src/main/java/io/sentry/android/replay/util/SystemProperties.kt

```diff
@@ -0,0 +1,34 @@
+package io.sentry.android.replay.util
+
+import android.annotation.SuppressLint
+import android.os.Build
+import java.lang.reflect.Method
+
+internal object SystemProperties {
+    // from https://cs.android.com/android/platform/superproject/main/+/main:out/soong/.intermediates/system/libsysprop/srcs/PlatformProperties/android_common/xref/srcjars.xref/android/sysprop/SocProperties.java;l=163-171
+    // these props are not available on API < 31 via Build, so we use reflection to access them
+    const val SOC_MODEL = "ro.soc.model"
+    const val SOC_MANUFACTURER = "ro.soc.manufacturer"
+
+    @delegate:SuppressLint("PrivateApi")
+    private val getProperty: Method by lazy {
+        val clazz = Class.forName("android.os.SystemProperties")
+        clazz.getMethod("get", String::class.java)
+    }
+
+    fun get(key: String, defaultValue: String = ""): String {
+        return if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) {
+            try {
+                getProperty.invoke(null, key) as? String ?: defaultValue
```

Comment:
I actually have a feeling this will always return an empty string on API < 31, see https://android.googlesource.com/platform/system/libsysprop/+/ba383e1deb52077aa4ae446a983bbe9bf5497a34%5E1..ba383e1deb52077aa4ae446a983bbe9bf5497a34/, so we might as well just drop it.

According to the web, the T606 processor was introduced later that year, so there's a good chance it's not used on devices with API < 31.

![image](https://github.com/user-attachments/assets/aab9a13e-7d34-42f5-b887-964c7d29890e)